cmd/relay/bgs/bgs.go

```diff
@@ -0,0 +1,1306 @@
+				return nil
+			})
+
+			cachedAccount.SetPDS(canonicalHost.ID)
+		}
+		return cachedAccount, nil
+	}
+
+	newAccount := Account{
+		Did: did,
+		PDS: canonicalHost.ID,
+	}
+
+	err = bgs.db.Transaction(func(tx *gorm.DB) error {
+		res := tx.Model(&models.PDS{}).Where("id = ? AND repo_count < repo_limit", canonicalHost.ID).Update("repo_count", gorm.Expr("repo_count + 1"))
+		if res.Error != nil {
+			return fmt.Errorf("failed to increment repo count for pds %q: %w", canonicalHost.Host, res.Error)
+		}
+		if terr := bgs.db.Create(&newAccount).Error; terr != nil {
+			bgs.log.Error("failed to create user", "did", newAccount.Did, "err", terr)
+			return fmt.Errorf("failed to create other pds user: %w", terr)
+		}
+		return nil
+	})
+	if err != nil {
+		bgs.log.Error("user create and pds inc err", "err", err)
+		return nil, err
+	}
+
+	bgs.userCache.Add(did, &newAccount)
+
+	return &newAccount, nil
+}
+
+// UpdateAccountStatus is the database portion of receiving a #account message
+func (bgs *BGS) UpdateAccountStatus(ctx context.Context, did string, status string) error {
+	ctx, span := tracer.Start(ctx, "UpdateAccountStatus")
+	defer span.End()
+
+	span.SetAttributes(
+		attribute.String("did", did),
+		attribute.String("status", status),
+	)
+
+	u, err := bgs.lookupUserByDid(ctx, did)
+	if err != nil {
+		return err
+	}
+
+	switch status {
+	case events.AccountStatusActive:
+		// Unset the PDS-specific status flags
+		if err := bgs.db.Model(Account{}).Where("id = ?", u.ID).Update("upstream_status", events.AccountStatusActive).Error; err != nil {
+			return fmt.Errorf("failed to set user active status: %w", err)
+		}
+		u.SetUpstreamStatus(events.AccountStatusActive)
+	case events.AccountStatusDeactivated:
+		if err := bgs.db.Model(Account{}).Where("id = ?", u.ID).Update("upstream_status", events.AccountStatusDeactivated).Error; err != nil {
+			return fmt.Errorf("failed to set user deactivation status: %w", err)
+		}
+		u.SetUpstreamStatus(events.AccountStatusDeactivated)
+	case events.AccountStatusSuspended:
+		if err := bgs.db.Model(Account{}).Where("id = ?", u.ID).Update("upstream_status", events.AccountStatusSuspended).Error; err != nil {
+			return fmt.Errorf("failed to set user suspension status: %w", err)
+		}
+		u.SetUpstreamStatus(events.AccountStatusSuspended)
+	case events.AccountStatusTakendown:
+		if err := bgs.db.Model(Account{}).Where("id = ?", u.ID).Update("upstream_status", events.AccountStatusTakendown).Error; err != nil {
+			return fmt.Errorf("failed to set user taken down status: %w", err)
+		}
+		u.SetUpstreamStatus(events.AccountStatusTakendown)
+		// TODO: set Account takedown in db? -- bolson 2025
+	case events.AccountStatusDeleted:
+		// TODO: tweak model to mark user deleted? -- bolson 2025
+		if err := bgs.db.Model(&Account{}).Where("id = ?", u.ID).UpdateColumns(map[string]any{
+			"tombstoned":      true,
```

Comment:
the `tombstoned` and `handle` columns of this table/model are gone now.

we should treat `deleted` the same as all the other non-active account statuses, especially now that the relay doesn't hold a mirror/archive. accounts can enter the `deleted` state and then exit it and become `active` again, according to the spec.